test_cycler.py

```diff
@@ -279,3 +279,19 @@ def test_starange_init():
     c2 = cycler('lw', range(3))
     cy = Cycler(list(c), list(c2), zip)
     assert_equal(cy, c + c2)
+
+
+def test_concat():
+    a = cycler('a', range(3))
+    for con, chn in zip(a.concat(a), chain(a, a)):
```

Comment:
I'd be inclined to test this with two different cyclers, the way this is tested it would be possible to completely ignore the ``other`` input and concatenate self with self (which of course, is not the behaviour we want).